bot/utils/persistent_scheduling.py

```diff
@@ -99,7 +99,8 @@ async def delete_all(self) -> None:
         self._reschedule_task.cancel()
         await self.cache.clear()
 
-        self._reschedule_task.add_done_callback(self._scheduler.cancel_all())
+        self._reschedule_task.cancel()
+        self._scheduler.cancel_all()
```

Comment:
Why is the cancel repeated twice? The way to do it with `add_done_callback` would've been to create a lambda that ignores the argument given.

However, while the exact mechanics of task cancellation still elude me, I'm not sure if the `add_done_callback` approach is even necessary; the exception is thrown on the very next cycle of the event loop.

Reply:
Yeah I don't know why I did that, removed in b8d58e0